mlflow/lightgbm.py

```diff
@@ -837,22 +849,56 @@ def infer_model_signature(input_example):
                 registered_model_name=registered_model_name,
             )
 
+        # Whether to automatically log the training dataset as a dataset artifact.
+        if _log_datasets:
+            try:
+                data = train_set.data
```

Comment:
If the user specifies `valid_sets` (datasets for validation), can we log information about them to MLflow Tracking too? The user may also specify `valid_names`, which we could use as the contexts of the datasets. Can you check if `valid_names` is required when `valid_sets` is specified?

Reply:
Per lightgbm docs, `valid_names` is not required if `valid_sets` is specified.